packages/forms-web-app/src/dynamic-forms/middleware/get-journey-response-for-lpa.js

```diff
@@ -20,10 +21,17 @@ module.exports = () => async (req, res, next) => {
 	}
 
 	try {
-		const dbResponse = await getQuestionResponse(appealType, encodedReferenceId);
-		result = new JourneyResponse(appealType, referenceId, dbResponse?.answers, dbResponse.LPACode);
+		const dbResponse = await apiClient.getLPAQuestionnaire(referenceId);
+		const convertedResponse = mapDBResponseToJourneyResponseFormat(dbResponse);
+		result = new JourneyResponse(
+			appealType,
+			referenceId,
+			convertedResponse,
+			dbResponse.AppealCase?.LPACode
+		);
 	} catch (err) {
 		logger.error(err);
+		await apiClient.postLPAQuestionnaire(referenceId);
```

Comment:
I think if we need to create the questionnaire when not found this should be more explicit. If the error is something else, we don't always want to send this post.

Reply:
See https://github.com/Planning-Inspectorate/appeal-planning-decision/pull/3404/commits/4e8b3884aaedee71aeabe624e45a1079a8a06b17